crates/biome_js_analyze/src/lint/style/use_consistent_type_definitions.rs

```diff
@@ -0,0 +1,247 @@
+use crate::JsRuleAction;
+use biome_analyze::{
+    Ast, FixKind, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::{Markup, MarkupBuf, markup};
+use biome_diagnostics::Severity;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsDeclarationClause, AnyTsType, JsSyntaxKind, TsConditionalType, TsIdentBinding,
+    TsInterfaceDeclaration, TsMappedType, TsObjectType, TsTypeAliasDeclaration,
+};
+use biome_rowan::{AstNode, BatchMutationExt};
+use biome_rule_options::use_consistent_type_definitions::{
+    ConsistentTypeDefinition, UseConsistentTypeDefinitionsOptions,
+};
+
+declare_lint_rule! {
+    /// Enforce type definitions to consistently use either `interface` or `type`.
+    ///
+    /// _TypeScript_ provides two different ways to define an object type: `interface` and `type`.
+    ///
+    /// This rule enforces consistent usage of either `interface` or `type` for object type definitions.
+    /// Consistent type definition styles, aside from improving code readability, help minimize cognitive load when developers
+    /// switch between different codebases or within a large codebase.
+    ///
+    /// ## Example
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// type Point = { x: number; y: number; };
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```ts
+    /// interface Point {
+    ///   x: number;
+    ///   y: number;
+    /// }
+    /// ```
+    ///
+    /// ## Options
```

Comment:
Here how you should write and test options inside the docs: https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#general-structure